pkgs/applications/science/electronics/kicad/libraries.nix

```diff
@@ -35,7 +35,7 @@ in
   templates = mkLib "templates" { };
   footprints = mkLib "footprints" { };
   packages3d = mkLib "packages3d" {
-    hydraPlatforms = []; # this is a ~1 GiB download, occupies ~5 GiB in store
+    meta.hydraPlatforms = []; # this is a ~1 GiB download, occupies ~5 GiB in store
```

Comment:
Since we have
https://github.com/NixOS/nixpkgs/blob/f241aa6910450c3a54c8cf5bf0c501f72e09dcef/pkgs/applications/science/electronics/kicad/default.nix#L12
this line should probably be removed.

Reply:
the goal of that line is to not have the >5GB of 3d models on hydra,
it exceeds the output limit and it's just a download - unpack - move operation

i tagged you on this because you wrote the original line and i was hoping you'd know if this fix is correct

Reply:
The goal is also to have the default `kicad` being built on Hydra without a failure. If we think that setting `meta.hydraPlatforms` will prevent Hydra from downloading packages3d, then it will also prevent `kicad` from being built there, which is not the best. This should is formally okay to merge for ZHF, of course.

Reply:
unfortunately i don't know how to achieve the desired effect.  
at least `kicad-base` should be built by hydra, but isn't because it's not user usable so i don't want it in all-packages.nix
everything else is just downloads and default.nix (`kicad` and `kicad-unstable`) just wraps base with libraries

if there was a way to make the sub-packages build on their own, i'd mark all but base package (and the 18n one (a direct dependency)) as hydraPlatforms = [];

as there is no build-time dependency on the content of `kicad-packages3d`, maybe there's another way of resolving this?

Reply:
You could try to use `recurseIntoAttrs` for this.

Reply:
i'm not sure what that is a response to, i assume making the sub-packages build.  
i would kinda expect `nix repl` to then have `kicad.libraries.packages3d`, which it doesn't